api/WebApi.ts

```diff
@@ -124,7 +130,19 @@ export class WebApi {
             this.options.ignoreSslError = !!global['_vsts_task_lib_skip_cert_validation'];
         }
 
-        this.rest = new rm.RestClient('vsts-node-api', null, [this.authHandler], this.options);
+        let userAgent: string;
+        const vstsNodeApiVersion = JSON.parse(fs.readFileSync('package.json', 'utf8')).version;
+        const osName = os.platform();
+        const osVersion = os.release();
+
+        if(requestSettings) {
+            userAgent = requestSettings.productName + '/' + requestSettings.productVersion + ' (vsts-node-api ' + vstsNodeApiVersion + '; ' + osName + ' ' + osVersion + ')';
+        }
+        else {
+            userAgent = 'vsts-node-api/' + vstsNodeApiVersion + ' (' + osName + ' ' + osVersion + ')';
```

Comment:
Let's pull the repo name out into a variable, and it's now azure-devops-node-api :)